sycl/source/detail/config.cpp

```diff
@@ -62,36 +63,70 @@ void readConfig() {
   }
 
   if (File.is_open()) {
-    // TODO: Use max size from macro instead of 256
-    char Key[MAX_CONFIG_NAME] = {0}, Value[256] = {0};
+    char Key[MAX_CONFIG_NAME] = {0}, Value[MAX_CONFIG_VALUE] = {0};
+    std::string BufString;
+    std::size_t Position = std::string::npos;
     while (!File.eof()) {
-      // Expected fromat:
+      // Expected format:
       // ConfigName=Value\r
       // ConfigName=Value
       // TODO: Skip spaces before and after '='
-      File.getline(Key, sizeof(Key), '=');
-      if (File.fail()) {
-        // Fail to process the line. Skip it completely and try next one.
-        // Do we want to restore here? Or just throw an exception?
+      std::getline(File, BufString);
+      if (File.fail() && !File.eof()) {
+        // Fail to process the line.
         File.clear(File.rdstate() & ~std::ios_base::failbit);
         File.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+        throw sycl::exception(
+            make_error_code(errc::runtime),
+            "An error occurred while attempting to read a line");
+      }
+      // Skip lines with a length = 0 | which have a comment | don't have "="
+      if ((BufString.length() == 0) ||
+          (BufString.find("#") != std::string::npos) ||
+          (BufString.find("=") == std::string::npos)) {
         continue;
       }
-      File.getline(Value, sizeof(Value), '\n');
-
-      if (File.fail()) {
-        // Fail to process the value while config name is OK. It's likely that
-        // value is too long. Currently just deal what we have got and ignore
-        // remaining characters on the line.
-        // Do we want to restore here? Or just throw an exception?
-        File.clear(File.rdstate() & ~std::ios_base::failbit);
-        File.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+      // Finding the position of '='
+      Position = BufString.find("=");
+      // Checking that the variable name is less than MAX_CONFIG_NAME and more
+      // than zero character
+      if ((Position <= MAX_CONFIG_NAME) && (Position > 0)) {
+        // Checking for spaces at the beginning and end of the line,
+        // before and after '='
+        if ((BufString[0] == ' ') ||
+            (BufString[BufString.length() - 1] == ' ') ||
+            (BufString[Position - 1] == ' ') ||
+            (BufString[Position + 1] == ' ')) {
+          throw sycl::exception(make_error_code(errc::runtime),
```

Comment:
Why this case causing the exception. Should we truncate all such spaces?

Reply:
Since we don't have SYCL config specification, I expect that assignment in config will be strict, without extra SPACE.

Reply:
lets leave it as is until TODO: in the comment below is fixed